src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -199,23 +214,42 @@ case class Block(center: (Double,Double,Double), faces: Seq[Face]) {
         val newY = NumericUtils.roundToTolerance(centerY + y)
         val newZ = NumericUtils.roundToTolerance(centerZ + z)
         val updatedFaces = updateFaces(newX, newY, newZ)
-        if (translatedJoint.d < 0.0) {
-          Vector(
-            // New origin is guaranteed to lie within joint, so initial d = 0
-            Block((newX,newY,newZ), Face((-translatedJoint.a, -translatedJoint.b, -translatedJoint.c), 0.0,
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces),
-            Block((newX,newY,newZ), Face((translatedJoint.a,translatedJoint.b,translatedJoint.c), 0.0, 
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
-          )
+
+        // New origin is guaranteed to lie within joint, so initial d = 0 for all child blocks
+        val childBlockA = if (translatedJoint.d < 0.0) {
+          Block((newX,newY,newZ), Face((-translatedJoint.a, -translatedJoint.b, -translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
+        } else {
+          Block((newX,newY,newZ), Face((translatedJoint.a, translatedJoint.b, translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
+        }
+        val childBlockB = if (translatedJoint.d < 0.0) {
+          Block((newX,newY,newZ), Face((translatedJoint.a,translatedJoint.b,translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
         } else {
-          Vector(
-            // New origin is guaranteed to lie within joint, so initial d = 0
-            Block((newX,newY,newZ), Face((translatedJoint.a, translatedJoint.b, translatedJoint.c), 0.0,
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces),
-            Block((newX,newY,newZ), Face((-translatedJoint.a,-translatedJoint.b,-translatedJoint.c), 0.0,
-                  translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
-          )
+          Block((newX,newY,newZ), Face((-translatedJoint.a,-translatedJoint.b,-translatedJoint.c), 0.0,
+            translatedJoint.phi, translatedJoint.cohesion)+:updatedFaces)
         }
+
+        var childBlocks = Vector(childBlockA, childBlockB)
```

Comment:
It seems like you could do this without having to use a `var`. 
Wouldn't it be better to just put explicit `return` values inside some conditionals: If it fails the first check, there's no need to do anything else and just return the original block. Then if it makes it all the way to the end without returning, you can just return `Vector(childBlockA, childBlockB)`. 
Would that be reasonable or am I missing something?